services/web/server/tests/unit/with_dbs/02/test_projects_service_conditionally_unsubscribe_from_project_logs.py

```diff
@@ -0,0 +1,109 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-positional-arguments
+# pylint: disable=too-many-statements
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+import contextlib
+from collections.abc import Awaitable, Callable
+from unittest import mock
+
+import pytest
+import socketio
+from aiohttp.test_utils import TestClient
+from common import SocketHandlers
+from pytest_mock import MockerFixture
+from pytest_simcore.helpers.webserver_login import log_client_in
+from pytest_simcore.helpers.webserver_parametrizations import (
+    ExpectedResponse,
+)
+from servicelib.aiohttp import status
+from simcore_service_webserver.db.models import UserRole
+from test_projects_states_handlers import _close_project, _open_project
+
+
+@pytest.fixture
+def max_number_of_user_sessions() -> int:
+    return 3
+
+
+@pytest.fixture
+def mock_publish_unsubscribe_from_project_logs_event(mocker: MockerFixture) -> None:
```

Comment:
1. inner import and `mocker.patch.object` is far more maintainable
2. fix return type
3. rename as `mocked` in order to distinguish from the `mock` factory (we have many out there!)

```python
from pytest_mock import MockType

def mocked_publish_unsubscribe_from_project_logs_event(mocker: MockerFixture) -> MockType:
    import simcore_service_webserver.projects._projects_service
    return mocker.patch.object(simcore_service_webserver.projects._projects_service,   "_publish_unsubscribe_from_project_logs_event",
        autospec=True,
    )
```